New header structure and style

andersnoren opened the pull request:
Updates the header structure and style to match the Twenty Twenty design. This commit also includes the new ”Shortcuts Menu”, which is displayed next to the header toggles on 1000px and up. We should have a discussion on what the most appropriate name for it is. I'll create an issue.

The commit also changes all toggles to be button elements instead of links (fixes #66), and changes the aria-pressed attribute to aria-expanded (fixes #68).

The commit does _not_ include accessible drop down menus for the shortcuts menu. I’m guessing we’ll use a JS based solution for that, in which case it might be better to wait until the jQuery → vanilla JS update is done.

pattonwebz commented on functions.php:
```diff
@@ -172,9 +172,10 @@ function twentytwenty_menus() {
-            'main-menu' 	=> __( 'Main Menu', 'twentytwenty' ),
-            'footer-menu' 	=> __( 'Footer Menu', 'twentytwenty' ),
-            'social-menu' 	=> __( 'Social Menu', 'twentytwenty' ),
+            'footer-menu' 		=> __( 'Footer Menu', 'twentytwenty' ),
```
One too many tabs?

andersnoren replied:
That there was! Fixed now, thanks :)

andersnoren commented:
Fixed the merge conflict. The admin bar fix is working on my local.

carolinan commented:
When the screen size is reduced and the search is moved to the left of the site title, the visual order does not correspond with the semantic order.

The focus is now title, menu, and then all the way back to the left side for the search.
The expected result is the same as the visual order: search, title, menu.
https://make.wordpress.org/themes/handbook/review/accessibility/required/#keyboard-navigation

@afercia would be able to explain this better than me.

carolinan commented:
I believe that the two svg icons should not be focusable, because it means two tabs for each item, first the button and then the icon, and they both open the same modals.

pattonwebz commented:
I think adding `aria-hidden="true"`, `focusable="false"` would prevent the svgs being double tab to pass through. May also need `role="presentational"` just for more context to screen readers.

andersnoren commented:
Could be the `tabIndex` not having a capitalized `i`. Taking a look.

I'll have to think a bit on how to update the structure to match the focus order. We might need a different set of toggles for mobile/portrait tablet (when they are to left and right of the site title).

andersnoren commented:
@carolinan I've added the focus fix and updated the header structure to include mobile and tablet specific toggles, which are hidden → 1000px when the regular toggles appear.

afercia commented:
> When the screen size is reduced and the search is moved to the left of the site title, the visual order does not correspond with the semantic order.

Thanks @carolinan I just noticed this issue looking at the design mockups on [the Make post](https://make.wordpress.org/core/2019/09/06/introducing-twenty-twenty/), which show a different design from what's in the theme at the moment. I was unaware the header design was going to change.

<img width="962" alt="header order" src="https://user-images.githubusercontent.com/1682452/64477662-95d03900-d19e-11e9-8662-7cd554c63e49.png">

Can't test this PR at the moment but looking at the screenshot above, the order changes from large to small screens. 

If the order is changed via CSS, that's an accessibility problem. Reading and focus order must always match visual order.

Relevant WCAG requirements:
1.3.2 Meaningful Sequence (Level A) https://www.w3.org/TR/WCAG20/#content-structure-separation-sequence
2.4.3 Focus Order (Level A) https://www.w3.org/TR/WCAG20/#navigation-mechanisms-focus-order

Icon-only controls are also a big problem. Will create a separate issue later.
